HelpersWayland.cpp

```diff
@@ -0,0 +1,295 @@
+#ifdef WAYLAND_PROTOCOL
+
+#include <iostream>
+
+#include "HelpersWayland.h"
+
+bool HelpersWayland::hasProperLibEGL() {
+    driverName = (eglGetDisplayDriverName_t *) eglGetProcAddress("eglGetDisplayDriverName");
+    driverConfig = (eglGetDisplayDriverconfig_t *) eglGetProcAddress("eglGetDisplayDriverConfig");
+
+    if (!driverName || !driverConfig) {
+        std::cerr << "Error getting function pointers. LibEGL must be too old.\n";
+        return false;
+    }
+
+    //std::cout << "System has proper LibEGL\n";
+    return true;
+}
+
+
+void HelpersWayland::error_fatal(const char* format, ...) {
+    printf("error: ");
+
+    va_list va;
+    va_start(va, format);
+    vprintf(format, va);
+    va_end(va);
+
+    printf("\n");
+    exit(1);
+}
+
+const uint32_t xcb_window_attrib_mask = XCB_CW_EVENT_MASK;
+const uint32_t xcb_window_attrib_list[] = {
+    XCB_EVENT_MASK_BUTTON_PRESS,
+    XCB_EVENT_MASK_EXPOSURE,
+    XCB_EVENT_MASK_KEY_PRESS,
+};
+
+/* This function is used to retrieve xcb_screen_t associated to
+ * particular Xlib-screen.
+ *
+ * ref: https://www.x.org/releases/X11R7.6/doc/libxcb/tutorial/index.html#screenofdisplay
+ * */
+xcb_screen_t *HelpersWayland::screen_of_display (xcb_connection_t *c,
+                                 int               screen)
+{
+  xcb_screen_iterator_t iter;
+
+  iter = xcb_setup_roots_iterator (xcb_get_setup (c));
+  for (; iter.rem; --screen, xcb_screen_next (&iter))
+    if (screen == 0)
+      return iter.data;
+
+  return NULL;
+}
+
+void HelpersWayland::setup_x(const char *display_name,
```

Comment:
It can happen that we have no X11 support at all on a pure-wayland system, therefore this code will always fail.
It should be completely independent of X11 (Xlib/xcb)